protocol/signalfx/additionalspantags/additionalspantags.go

```diff
@@ -0,0 +1,54 @@
+package additionalspantags
+
+import (
+	"context"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/datapoint/dpsink"
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/golib/trace"
+)
+
+type sink interface {
+	dpsink.Sink
+	trace.Sink
+}
+
+var _ sink = &AdditionalSpanTags{}
+
+// AdditionalSpanTags adds tags to every span it processes
+type AdditionalSpanTags struct {
+	tags map[string]string
+	next sink
+}
+
+// AddDatapoints is a passthrough
+func (a *AdditionalSpanTags) AddDatapoints(ctx context.Context, points []*datapoint.Datapoint) error {
+	return a.next.AddDatapoints(ctx, points)
+}
+
+// AddEvents is a passthrough
+func (a *AdditionalSpanTags) AddEvents(ctx context.Context, events []*event.Event) error {
+	return a.next.AddEvents(ctx, events)
+}
+
+// AddSpans adds tags to all of the spans that are given
+func (a *AdditionalSpanTags) AddSpans(ctx context.Context, spans []*trace.Span) error {
+	for _, s := range spans {
+		if s.Tags == nil {
+			s.Tags = map[string]string{}
```

Comment:
benchmark this vs make(map[string]string, len(a.tags))

Reply:
The performance is a little better with make(), but very close when using we're only adding two tags
`$benchcmp empty_map make_len_tags-two 
benchmark        old ns/op     new ns/op     delta
Benchmark-12     437           404           -7.55%

benchmark        old allocs     new allocs     delta
Benchmark-12     2              2              +0.00%

benchmark        old bytes     new bytes     delta
Benchmark-12     336           336           +0.00%`

There is a large performance jump if we're adding a large number of tags, e.g 100
`$benchcmp empty_map-hundred make_len_tags-hundred 
benchmark        old ns/op     new ns/op     delta
Benchmark-12     11694         8211          -29.78%

benchmark        old allocs     new allocs     delta
Benchmark-12     11             4              -63.64%

benchmark        old bytes     new bytes     delta
Benchmark-12     10515         5457          -48.10%`

Either way, make() is the better option